modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java

```diff
@@ -228,7 +228,22 @@ protected boolean isMember(
 		Criteria.Conjunction contextConjunction = getConjunction(
 			segmentsEntry, Criteria.Type.CONTEXT);
 
+		String modelFilterString = getFilterString(
+			segmentsEntry, Criteria.Type.MODEL);
+
 		if (context != null) {
+			boolean guestUser = false;
+
+			if (context.containsKey(Context.SIGNED_IN) &&
+				!GetterUtil.getBoolean(context.get(Context.SIGNED_IN))) {
+
+				guestUser = true;
+			}
+
+			if (guestUser && Validator.isNotNull(modelFilterString)) {
+				return false;
```

Comment:
Notice that in case the conjunction is an OR is going to return false negative!

Reply:
Good catch! I didn't notice the "OR" case.